tools/assetprocessor/main.cpp

```diff
@@ -159,14 +225,38 @@ int main(int argc, char *argv[])
             return 1;
         }
 
-        gltfOut.write(QJsonDocument{ new_asset.json() }.toJson());
+        processedSize += gltfOut.write(newAssetJson);
     }
 
     // Show some stats
-    const auto space_taken = QFileInfo(targetDir.filePath(new_asset.compressedBufferFilename())).size();
-    QTextStream out(stdout);
-    out << QObject::tr("Kuesa: Compressing mesh took %1 milliseconds\n").arg(time_taken / 1000000.);
-    out << QObject::tr("Kuesa: Space taken by compressed meshes: %2 bytes\n").arg(space_taken);
+    if (stats) {
+        SceneEntity scene;
+        GLTF2Context ctx;
+        GLTF2Parser parser(&scene);
+        parser.setContext(&ctx);
+
+        timer.restart();
+        parser.parse(asset);
```

Comment:
aren't you parsing the same thing again?

Reply:
oops